crypto/params.c

```diff
@@ -1009,57 +1009,61 @@ int OSSL_PARAM_set_double(OSSL_PARAM *p, double val)
             *(double *)p->data = val;
             return 1;
         }
-    } else if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER
-               && val == (ossl_uintmax_t)val) {
-        p->return_size = sizeof(double);
+    } else if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER) {
+        if (val >= 0) {
+            if (val <= UINT32_MAX)
+                p->return_size = sizeof(uint32_t);
+            else if (val >= 0
+                     /*
+                      * By subtracting 65535 (2^16-1) we cancel the low order
```

Comment:
Hmmmm....  this will work but I'm not sure its the best way to go when converting real -> uint.  This approach is fine for converting uint -> real safely.

Reply:
The more I think about this, the wronger it seems.